### PR TITLE
Rds/jdj/rule5 43

### PR DESCRIPTION
**Pull request by JacksonJ-KC**

### CODE REVIEW

**Comment by JacksonJ-KC** [comment]
> I thought these rules were decided to be primary rules?

@weilixu yes they were, ignore or feel free to change the file names for these rules. The file naming was temporary until the previous RDS are archived/overwritten as needed. (I did not want to make the decision to overwrite the previous files without input from you or the group if needed)

**Comment by weilixu** [comment]
> > I thought these rules were decided to be primary rules?
> 
> @weilixu (5-29, 5-41, 5-43, 5-53) yes they were, ignore or feel free to change the file names for these rules. The file naming was temporary until the previous RDS are archived/overwritten as needed. (I did not want to make the decision to overwrite the previous files without input from you or the group if needed)

Hi Jackson, those rules were previously secondary/non mandatory rules. 
I am a little confused now. I thought, we decided to change these rules to primary/mandatory rules. Is that correct?
I do see the existing RDSs were drafted and merged with primary rule logic.

**Comment by JacksonJ-KC** [comment]
> > > I thought these rules were decided to be primary rules?
> > 
> > 
> > @weilixu (5-29, 5-41, 5-43, 5-53) yes they were, ignore or feel free to change the file names for these rules. The file naming was temporary until the previous RDS are archived/overwritten as needed. (I did not want to make the decision to overwrite the previous files without input from you or the group if needed)
> 
> Hi Jackson, those rules were previously secondary/non mandatory rules. I am a little confused now. I thought, we decided to change these rules to primary/mandatory rules. Is that correct? I do see the existing RDSs were drafted and merged with primary rule logic.

Sorry for the confusion! Those 4 rules existing RDS were drafted with primary rule logic but were at some point changed to nonprimary rules before I joined. Then I joined, I began drafting new RDS for all nonprimary rules, and appended "_nonprimary" to the file names until I knew what we wanted to do with the existing RDS files. Then these 4 rules were changed to back to primary. This pull request still has the file name "_nonprimary" but is really my attempt at updating the primary RDS. I knew the "_nonprimary" file name was temporary anyways so I did not update it, but I should have done so to avoid confusion.

File: docs/section5/Rule5-43_nonprimary.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Envelope - Rule 5-43  
+**Schema Version** 0.0.23  
+**Primary Rule:** False
```
Comment (supriyagoel) [inline docs/section5/Rule5-43_nonprimary.md:3]: changee primary rule to true

File: docs/section5/Rule5-43_nonprimary.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - For each zone in building segment: ```for zone_p in building_segment_p.zones:```
+    
+    - For each surface in zone: ```for surface_p in zone_p.surfaces:```
+
+      - Check if surface is roof, set rule applicability check to True: ```if get_opaque_surface_type(surface_p.id) == "ROOF": rule_applicability_check = TRUE```
+  
+        - Get matching surface in U_RMR: ```surface_u = match_data_element(U_RMR, Surfaces, surface_p.id)```
+        **Rule Assertion:**
+
+        - Case 1: If roof surface solar reflectance in P_RMR matches U_RMR and is equal to 0.3, outcome is PASS: ```if ( surface_p.surface_optical_properties.absorptance_solar_exterior == surface_u.surface_optical_properties.absorptance_solar_exterior ) AND ( surface_p.surface_optical_properties.absorptance_solar_exterior == 0.7 ):
```
Comment (supriyagoel) [inline docs/section5/Rule5-43_nonprimary.md:33]: would be helpful to have a clarifying note here than solar reflectance = 1-solar absorptance.

File: docs/section5/Rule5-43_nonprimary.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          outcome = UNDETERMINED and raise_message "ROOF SURFACE SOLAR REFLECTANCE IN P-RMR (${1 - surface_p.surface_optical_properties.absorptance_solar_exterior}) MATCHES THAT IN U-RMR BUT IS NOT EQUAL TO THE PRESCRIBED DEFAULT VALUE OF 0.3. VERIFY THAT REFLECTANCE WAS ESTABLISHED USING AGED TEST DATA AS REQUIRED IN SECTION 5.5.3.1(a).```
+
+        - Case 3: Else if roof surface solar reflectance in P_RMR does not match U_RMR but is equal to 0.3, outcome is UNDETERMINED: ```else if surface_p.surface_optical_properties.absorptance_solar_exterior == 0.7:
+          outcome = UNDETERMINED and raise_message "ROOF SURFACE SOLAR REFLECTANCE IS EQUAL TO THE PRESCRIBED DEFAULT VALUE OF 0.3 BUT DIFFERS FROM THE SOLAR REFLECTANCE IN THE USER MODEL (${1 - surface_u.surface_optical_properties.absorptance_solar_exterior})."```
+
+        - Case 4: Else, roof surface solar reflectance in P_RMR does not match that in U_RMR and is not equal to 0.3, outcome is FAIL: ```Else: outcome = FAIL```
+
+- Case 5: If no surface in building is roof, outcome is NOT_APPLICABLE: ```if NOT rule_applicability_check: outcome = NOT_APPLICABLE```
```
Comment (supriyagoel) [inline docs/section5/Rule5-43_nonprimary.md:44]: not applicability check is not required here. that is checked in line 28

**Review by supriyagoel — APPROVED** [review]

File: docs/section5/Rule5-43_nonprimary.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        
+        - Case 2: Else if roof surface solar reflectance in P_RMR matches U_RMR and is not equal to 0.3, outcome is UNDETERMINED: ```else if ( surface_p.surface_optical_properties.absorptance_solar_exterior == surface_u.surface_optical_properties.absorptance_solar_exterior ) AND ( surface_p.surface_optical_properties.absorptance_solar_exterior != 0.7 ):
+          outcome = UNDETERMINED and raise_message "ROOF SURFACE SOLAR REFLECTANCE IN P-RMR (${1 - surface_p.surface_optical_properties.absorptance_solar_exterior}) MATCHES THAT IN U-RMR BUT IS NOT EQUAL TO THE PRESCRIBED DEFAULT VALUE OF 0.3. VERIFY THAT REFLECTANCE WAS ESTABLISHED USING AGED TEST DATA AS REQUIRED IN SECTION 5.5.3.1(a).```
+
+        - Case 3: Else if roof surface solar reflectance in P_RMR does not match U_RMR but is equal to 0.3, outcome is UNDETERMINED: ```else if surface_p.surface_optical_properties.absorptance_solar_exterior == 0.7:
+          outcome = UNDETERMINED and raise_message "ROOF SURFACE SOLAR REFLECTANCE IS EQUAL TO THE PRESCRIBED DEFAULT VALUE OF 0.3 BUT DIFFERS FROM THE SOLAR REFLECTANCE IN THE USER MODEL (${1 - surface_u.surface_optical_properties.absorptance_solar_exterior})."```
```
Comment (mkarpman) [inline docs/section5/Rule5-43_nonprimary.md:42]: I would consider making it a PASS but keeping the message. @supriyagoel do you agree?

**Review by mkarpman — APPROVED** [review]

**Pull request merged**